src/main/java/hw02/Family.java

```diff
@@ -58,4 +61,54 @@ public Pet getPet() {
     public void setPet(Pet pet) {
         this.pet = pet;
     }
+
+    public boolean deleteChild(Human child) {
+        boolean isDeleted = false;
+        for (int i = 0; i < children.length; i++) {
+            if (children[i].equals(child)) {
+                children[i] = null;
+                isDeleted = true;
+            }
+        }
+        return isDeleted;
+    }
+
+    public Human[] addChild(Human child) {
+        Human[] newChildren = Arrays.copyOf(children, children.length + 1);
+        newChildren[children.length] = child;
+        children = newChildren;
+        return children;
+    }
+
+    public int countFamily() {
+        return children.length + 2;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(father, mother);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Family that = (Family) obj;
+        return Objects.equals(father, that.father) &&
+                Objects.equals(mother, that.mother);
+    }
+
+    @Override
+    public String toString() {
+        return "Family{" +
+                "mother=" + (mother != null ? mother.toString() : "none") +
```

Comment:
Family don't have default constructor, these means for creating Family object mother and father objects are requeired, then no need this check here